lib/src/css_parser.dart

```diff
@@ -70,6 +115,266 @@ class ExpressionMapping {
+        case "line-through":
+          decorationList.add(TextDecoration.lineThrough);
+          break;
+        default:
+          decorationList.add(TextDecoration.none);
+          break;
+      }
+    }
+    return TextDecoration.combine(decorationList);
+  }
+
+  static TextDecorationStyle expressionToTextDecorationStyle(css.LiteralTerm value) {
+    switch(value.text) {
+      case "wavy":
+        return TextDecorationStyle.wavy;
+      case "dotted":
+        return TextDecorationStyle.dotted;
+      case "dashed":
+        return TextDecorationStyle.dashed;
+      case "double":
+        return TextDecorationStyle.double;
+      default:
+        return TextDecorationStyle.solid;
+    }
+  }
+
+  static List<Shadow> expressionToTextShadow(List<css.Expression> value) {
+    List<Shadow> shadow = [];
+    List<int> indices = [];
+    List<List<css.Expression>> valueList = [];
+    for (css.Expression e in value) {
+      if (e is css.OperatorComma) {
+        indices.add(value.indexOf(e));
+      }
+    }
+    indices.add(value.length);
+    int previousIndex = 0;
+    for (int i in indices) {
+      valueList.add(value.sublist(previousIndex, i));
+      previousIndex = i + 1;
+    }
+    for (List<css.Expression> list in valueList) {
+      css.Expression exp = list[0];
+      css.Expression exp2 = list[1];
+      css.LiteralTerm exp3 = list.length > 2 ? list[2] : null;
+      css.LiteralTerm exp4 = list.length > 3 ? list[3] : null;
+      RegExp nonNumberRegex = RegExp(r'\s+(\d+\.\d+)\s+');
+      if (exp is css.LiteralTerm && exp2 is css.LiteralTerm) {
+        if (exp3 != null && (exp3 is css.HexColorTerm || exp3 is css.FunctionTerm)) {
+          shadow.add(Shadow(
+              color: expressionToColor(exp3), 
+              offset: Offset(double.tryParse(exp.text.replaceAll(nonNumberRegex, '')), double.tryParse(exp2.text.replaceAll(nonNumberRegex, '')))
+          ));
+        } else if (exp3 != null && exp3 is css.LiteralTerm) {
+          if (exp4 != null && (exp4 is css.HexColorTerm || exp4 is css.FunctionTerm)) {
+            shadow.add(Shadow(
+                color: expressionToColor(exp4), 
+                offset: Offset(double.tryParse(exp.text.replaceAll(nonNumberRegex, '')), double.tryParse(exp2.text.replaceAll(nonNumberRegex, ''))), 
+                blurRadius: double.tryParse(exp3.text.replaceAll(nonNumberRegex, ''))
+            ));
+          } else {
+            shadow.add(Shadow(
+                offset: Offset(double.tryParse(exp.text.replaceAll(nonNumberRegex, '')), double.tryParse(exp2.text.replaceAll(nonNumberRegex, ''))), 
+                blurRadius: double.tryParse(exp3.text.replaceAll(nonNumberRegex, ''))
+            ));
+          }
+        } else {
+          shadow.add(Shadow(
+              offset: Offset(double.tryParse(exp.text.replaceAll(nonNumberRegex, '')), double.tryParse(exp2.text.replaceAll(nonNumberRegex, '')))
+          ));
+        }
+      }
+    }
+    shadow.toSet().toList();
```

Comment:
Same here - this has no effect right?

Reply:
It removes duplicates from the list here as well.

Reply:
Fixed.